### DISCUSSION TITLE
Error code: 401101, "Not Authorized to access the API"

### DISCUSSION THREAD

**Discussion by kibiwotvincent** [njoguamos/laravel-jenga#43] [Q&A] [answered]
I have cloned the repo and followed all the steps in generating and uploading the public key to my jenga account. I have doubled checked on my API key, consumer secret and merchant code - they are all correct. I can confirm that my access token and signature are correct.
So what I wanted is to do a simple account search before I could dig into the API deeply just to make sure I have everything running correctly.
Here is my web route:

```
<?php

use Illuminate\Support\Facades\Route;
use App\Http\Controllers\AccountInquiryController;

/*
|--------------------------------------------------------------------------
| Web Routes
|--------------------------------------------------------------------------
|
| Here is where you can register web routes for your application. These
| routes are loaded by the RouteServiceProvider and all of them will
| be assigned to the "web" middleware group. Make something great!
|
*/


Route::get('/search', [AccountInquiryController::class, 'search']);

```
and here is my AccountInquiryController:

```
<?php

namespace App\Http\Controllers;

```
use Illuminate\Http\Request;
use NjoguAmos\Jenga\Api\AccountInquiry;

class AccountInquiryController extends Controller
{

    public function search(Request $request)
    {
        /*$validated = $request->validate([
            'countryCode' => ['required'],
            'accountNumber' => ['required'],
        ]);*/
        
        $validated['countryCode'] = 'KE';
        $validated['accountNumber'] = '1450160649886';

        $response = (new AccountInquiry())
            ->search(
                countryCode: $validated['countryCode'],
                accountNumber: $validated['accountNumber']
            );
        
        print $response;
    }
    
}

```
```
As you can see I am doing a search on the default sandbox account.
The result I am getting is:

`{"status":false,"code":401101,"message":"Not Authorized to access the API"}`

My issue is, if all my credentials are right, why am I still not authorized into the API?
I have tested with my real Equity bank account but the error still persists. Are there steps that I am missing?

**Comment by njoguamos** [comment, marked as answer]
@kibiwotvincent If everything is correct, as you have suggested, then there is a possibility that you have not subscribed to the respective service. In JengaAPI, you have to enable the API services that you need to use. This is security measure.


Start by going to Settings -> Subscriptions and ensure your account is subscribed. 


<img width="1159" alt="Screenshot 2023-12-07 at 22 21 35" src="https://github.com/njoguamos/laravel-jenga/assets/29255728/94b4c23e-2001-4ef6-bc7c-dfdcaa560468">

**Reply by kibiwotvincent** [reply]
This works. Thank you very much. This is such a crucial piece of information to be left out in the JengaAPI docs.

**Reply by njoguamos** [reply]
I am glad it worked. I have updated the [docs](https://github.com/njoguamos/laravel-jenga/releases/tag/v1.5.1)
